src/circuit/relay.ts

```diff
@@ -0,0 +1,108 @@
+import { logger } from '@libp2p/logger'
+import { codes } from '../errors.js'
+import {
+  setDelayedInterval,
+  clearDelayedInterval
+// @ts-expect-error set-delayed-interval does not export types
+} from 'set-delayed-interval'
+import { namespaceToCid } from './utils.js'
+import {
+  RELAY_RENDEZVOUS_NS
+} from './constants.js'
+import type { AddressSorter } from '@libp2p/interface-peer-store'
+import type { Startable } from '@libp2p/interfaces/startable'
+import type { Components } from '../components.js'
+
+const log = logger('libp2p:relay')
+
+export interface RelayAdvertiseConfig {
+  bootDelay?: number
+  enabled?: boolean
+  ttl?: number
+}
+
+export interface HopConfig {
+  enabled?: boolean
+  active?: boolean
+}
+
+export interface AutoRelayConfig {
```

Comment:
Do we still use this? We should either remove or rename it

Reply:
We use it to automatically initiate a reservation on newly connected peers or if the circuit relay protocol is added to an existing peer.